src/main/kotlin/no/nav/syfo/identhendelse/IdenthendelseService.kt

```diff
@@ -0,0 +1,74 @@
+package no.nav.syfo.identhendelse
+
+import kotlinx.coroutines.runBlocking
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.client.pdl.PdlClient
+import no.nav.syfo.dialogmotekandidat.database.getDialogmotekandidatEndringListForPerson
+import no.nav.syfo.dialogmotekandidat.database.getDialogmotekandidatStoppunktList
+import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.identhendelse.database.*
+import no.nav.syfo.identhendelse.kafka.COUNT_KAFKA_CONSUMER_PDL_AKTOR_UPDATES
+import no.nav.syfo.identhendelse.kafka.KafkaIdenthendelseDTO
+import no.nav.syfo.unntak.database.getUnntakList
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class IdenthendelseService(
+    private val database: DatabaseInterface,
+    private val pdlClient: PdlClient,
+) {
+
+    private val log: Logger = LoggerFactory.getLogger(IdenthendelseService::class.java)
+
+    fun handleIdenthendelse(identhendelse: KafkaIdenthendelseDTO) {
+        if (identhendelse.folkeregisterIdenter.size > 1) {
+            val activeIdent = identhendelse.getActivePersonident()
+            if (activeIdent != null) {
+                val inactiveIdenter = identhendelse.getInactivePersonidenter()
+                val numberOfUpdatedIdenter = updateAllTables(activeIdent, inactiveIdenter)
+                log.info("Identhendelse: Updated $numberOfUpdatedIdenter rows based on Identhendelse from PDL")
+                COUNT_KAFKA_CONSUMER_PDL_AKTOR_UPDATES.increment(numberOfUpdatedIdenter.toDouble())
+            } else {
+                log.warn("Mangler gyldig ident fra PDL")
+            }
+        }
+    }
+
+    private fun updateAllTables(activeIdent: PersonIdentNumber, inactiveIdenter: List<PersonIdentNumber>): Int {
+        var numberOfUpdatedIdenter = 0
+        val oldStoppunktList = inactiveIdenter.flatMap { database.getDialogmotekandidatStoppunktList(it) }
```

Comment:
Kanskje dette blir enklere dersom det er en loop på inactiveIdenter, og så en update(activeIdent, oldIdent) alle steder der oldIdent forekommer (med select count(id)..)

Reply:
Hmm, problemet her blir at vi er litt låst til rekkefølgen:
1. sjekke om vi har noe i databasen
2. så gjøre sjekk mot pdl (helst gjøres bare én gang)
3. gjøre updates

Punkt 2 burde ikke loopes 🤔

Reply:
Hadde vært nice om det var mulig å gjøre én sjekk på om vi har noe i basen, deretter sjekket pdl, og så loopet inactive-identer og gjort update i alle tabeller (gjør vel ikke noe om man forsøker update en tabell hvor inactive ident ikke finnes?) 🤔

Reply:
Mulig man kunne gjort spørringen i én query mot alle tabellene, typ SELECT COUNT med UNION eller noe sånt. @geir-waagboe vet kanskje? I så fall kunne det forenklet koden noe i service-klassen. Hvis man uansett bare trenger å vite om identen finnes ett sted og så oppdatere alle.

Reply:
Kunne gjort noe som dette da. Evt at man bytter ut update-funksjonene til å bare ta inn én ident og ikke hele lista, og dermed heller loope. Det er vel kanskje bedre å sende inn en liste og gjenbruke connection, enn å starte en ny connection mot alle databasene for hver entry i `inactiveIdenter`

```kotlin
val rowsWithInactiveIdenter = inactiveIdenter.sumOf {
            database.getDialogmotekandidatStoppunktList(it).size +
                database.connection.getDialogmotekandidatEndringListForPerson(it).size +
                database.getUnntakList(it).size +
                database.getDialogmoteStatusCount(it)
        }
        
        if (rowsWithInactiveIdenter > 0) {
            checkThatPdlIsUpdated(activeIdent)
            numberOfUpdatedIdenter += database.updateKandidatStoppunkt(activeIdent, inactiveIdenter)
            numberOfUpdatedIdenter += database.updateKandidatEndring(activeIdent, inactiveIdenter)
            numberOfUpdatedIdenter += database.updateUnntak(activeIdent, inactiveIdenter)
            numberOfUpdatedIdenter += database.updateDialogmoteStatus(activeIdent, inactiveIdenter)
        }
    
        return numberOfUpdatedIdenter
```

Reply:
Aha, ja hvis man kunne spurt alle tabeller i én sleng hadde jo det vært kult å prøve 👍🏼

Reply:
Vil tro vi ønsker å gjenbruke connection, ja.